.github/workflows/publish_develop.yml

```diff
@@ -52,9 +52,53 @@ jobs:
             TAG=$TAG-$SHORT_SHA
           fi
           echo "TAG=$TAG" >> $GITHUB_ENV
-      - name: Build and push image
+      - name: Build and run image
         run: |
+          pip install -r requirements.txt
           docker build -t ${{ secrets.AWS_ECR_DEV_REPO }} .
+          docker run -d -p 48008:8080 ${{ secrets.AWS_ECR_DEV_REPO }}
+          echo 'Creating a test dispatch...'
+          cat << EOF > test_dispatch.py
+          import covalent as ct
+          import time
+          from requests import request
+          import covalent_dispatcher._cli.service as service
+          @ct.electron
+          def join_words(a, b):
+            return ", ".join([a, b])
+          @ct.electron
+          def excitement(a):
+            return f"{a}!"
+          @ct.lattice
+          def simple_workflow(a, b):
+            phrase = join_words(a, b)
+            return excitement(phrase)
+          if service._is_server_running():
+            print('Dispatch service has started!')
+          else:
+            print('Waiting for dispatch to be executed...')
+            time.sleep(30)
+          dispatch_id = ct.dispatch(simple_workflow, dispatcher_addr='localhost:48008')("Hello", "Covalent")
+          results_url = "http://localhost:48008/api/results"
+          results = request("GET", results_url, headers={}, data={}).json()
+          dispatch_result = results[0]['result'] if results else None
+          dispatch_status = results[0]['status'] if results else None
+          print(f'Dispatch {dispatch_id} was executed successfully with status: {dispatch_status}')
+          print(f'The result of the dispatch is: {dispatch_result}')
+          EOF
+          cat << EOF > run_dispatch.py
+          import subprocess
+          import sys
+          for idx in range(2):
+            sys.stdout = open('result%s.txt' % idx, 'w')
+            subprocess.check_call(['python3', 'test_dispatch.py'],
+                          stdout=sys.stdout, stderr=subprocess.STDOUT)
+          EOF
+          python3 run_dispatch.py
+          cat result1.txt
+      - name: Push image
+        run: |
           docker push ${{ secrets.AWS_ECR_DEV_REPO }}
```

Comment:
We need to confirm that this push actually uses the container that was built in the previous step.